### PR TITLE
Failing APIv3 unit test.

### PR DESCRIPTION
**Pull request by colemanw**
tldr;
------
APIv3 is broken, but I think it never worked.

Description
-----------
It was pointed out in https://github.com/systopia/de.systopia.eck/issues/16#issuecomment-1054132747 that APIv3 was broken by the addition of an underscore to the entity name. There were no unit tests before to ensure it was working, so I've added one here. To my surprise, I found that the test fails on all versions, even before I refactored the api name with underscores. Action `get` works, but `create` and `delete` do not work.

Looking through the code, it looks like it was always broken. The `Entity::invoke` function only handles the `get` action:
https://github.com/systopia/de.systopia.eck/blob/d0e9e94bef5dbed8fe86cf609a103228ae893243/Civi/Eck/API/Entity.php#L134-L139

And although the UI calls `create`, it appears to have been a WIP because it's got a debug statement next to it.
https://github.com/systopia/de.systopia.eck/blob/d0e9e94bef5dbed8fe86cf609a103228ae893243/CRM/Eck/CustomData.php#L264-L266

Conclusion
-------------
If APIv3 was always broken, then I don't think there's any good reason to keep it. There was concern about keeping it for the sake of the rest api, but my response to that is 1) it's broken, and 2) the rest api now works with v4.

My recommendation would be to reopen #14

### CODE REVIEW

File: tests/phpunit/api/v3/EckEntityTypeTest.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    ]);
+    $this->assertTrue(is_numeric($entityType['id']));
+
+    $get = $this->callAPISuccess('EckEntityType', 'get', ['name' => 'TestAPIv3']);
+    $this->assertEquals(1, $get['count']);
+
+    // Table should have been created
+    $this->assertTrue(\CRM_Core_DAO::checkTableExists('civicrm_eck_testapiv3'));
+
+    $records = $this->callAPISuccess('EckTestAPIv3', 'get');
```
Comment (colemanw) [inline tests/phpunit/api/v3/EckEntityTypeTest.php:44]: You can run this test on a previous revision prior to #20 and it will get this far, but then fail on the `create` api call below. That's why I think `create` was always broken.

**Comment by jensschuppe** [comment]
Thanks for the text, @colemanw! You're right, only `get` (and `getfields` obviously) was working for now, but isn't anymore after the naming change. There are still APIv3 calls in this extension (`get` and `getfields` - the latter of which I noticed doesn't work with APIv4 yet), let's get rid of them first and work towards removing APIv3, which I agree with in general, but maybe release a version with APIv3 still in and without the naming change before doing that.

**Comment by colemanw** [comment]
Sounds good. FYI APIv4 `getFields` is working, and has the following test coverage:
https://github.com/systopia/de.systopia.eck/blob/a4fb482a67ca61f0b733663cfacd0884c513fcb0/tests/phpunit/api/v4/EckEntity/EckEntityTest.php#L179-L201

Everything works except the last line shows it can't filter custom fields by entity sub-type. IMO that's a core issue and is closely related to the problem of needing to store a pseudoconstant callback in the 'description' field of an option value. In both cases, better metadata about the relationship between sub-types and custom fields is needed.

**Comment by colemanw** [comment]
I've reopened #14
